docs/source/guides/adaptive_batching.rst

```diff
@@ -3,4 +3,44 @@
 Adaptive Batching
 =================
 
-TODO
+Batching is the term used for combining multiple inputs for submission to processing at the same time. The idea is that processing multiple messages is be faster than processing each individual message one at a time. In practice many ML frameworks have optimizations for processing multiple messages at a time because that how the underlying hardware works in many cases.
```

Comment:
```suggestion
Batching is the term used for combining multiple inputs for submission to processing at the same time. The idea is that processing multiple messages is be faster than processing each individual message one at a time. In practice many ML frameworks have optimizations for processing multiple messages at a time because that is how the underlying hardware works in many cases.
```